core/src/main/java/org/apache/struts2/dispatcher/HttpParameters.java

```diff
@@ -100,7 +112,7 @@ public boolean containsValue(Object value) {
 
     @Override
     public Parameter get(Object key) {
-        if (parameters.containsKey(key)) {
+        if (key != null && contains(String.valueOf(key))) {
```

Comment:
Not sure this part makes sense, checking for the existence of String#valueOf then retrieving using the object?

Do you mean something like this:
```
@Override
public Parameter get(Object key) {
    if (key == null) {
        return new Parameter.Empty("null");
    }
    Parameter val = parameters.get(key.toString());
    return val != null ? val : new Parameter.Empty(key.toString());
}
```

Reply:
Simplified to return `null` - no idea what was the reasoning behind :D

Reply:
I will use your code and I will add missing equals&hashCode